### PR TITLE
Developer template request

### PR DESCRIPTION
**Pull request by Freymaurer**
Please review this in depth. It will start a important part of ARCtrl logic and i would really like some input.

### CODE REVIEW

File: src/ARCtrl/Templates/Templates.Json.fs [hunk 1]
```diff
@@ -102,4 +102,19 @@ module Template =
...
             )
-        )
\ No newline at end of file
+        )
+
+[<AutoOpen>]
+module Extension =
+
+    type Template with
+        member this.ToJson(?spaces: int) =
```
Comment (HLWeil) [inline src/ARCtrl/Templates/Templates.Json.fs:111]: Stick to this naming?
https://github.com/nfdi4plants/ARCtrl/blob/14f173fea2b1efd44e26a372644800eb222f4dbd/src/ISA/ISA.Json/Study.fs#L112

File: src/ARCtrl/WebRequest/WebRequest.fs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+﻿module ARCtrl.WebRequest
+
+open Fable.Core
+open Fable.SimpleHttp
+open Fable.Core.JsInterop
+
+open Fable.SimpleHttp 
+
+[<RequireQualifiedAccess>]
+module MyNodeJs =
```
Comment (HLWeil) [inline src/ARCtrl/WebRequest/WebRequest.fs:10]: module name?

Comment (HLWeil) [reply src/ARCtrl/WebRequest/WebRequest.fs:10]: Or could you shortly explain what this module is for?

File: tests/TestingUtils/TestObjects.Json/MinimalJson.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"TestKey": "TestValue"}
```
Comment (HLWeil) [inline tests/TestingUtils/TestObjects.Json/MinimalJson.json:1]: Cute

**Review by HLWeil — CHANGES_REQUESTED** [review]

File: src/ARCtrl/Templates/Templates.Json.fs (hunk 1 above)
Comment (Freymaurer) [reply src/ARCtrl/Templates/Templates.Json.fs:111]: Sorry but `toString` for json sounds like a really bad idea here. I would accept changing `ofJson` to `fromJson`

File: src/ARCtrl/WebRequest/WebRequest.fs (hunk 2 above)
Comment (Freymaurer) [reply src/ARCtrl/WebRequest/WebRequest.fs:10]: This module contains js fable logic which only applied in a nodejs environment. 

This complete file should be restructured, but i am not sure what the best approach is. 

Maybe i should add a "Web" type containing requests as static members.

File: src/ARCtrl/Templates/Templates.Json.fs (hunk 1 above)
Comment (HLWeil) [reply src/ARCtrl/Templates/Templates.Json.fs:111]: Some name unification IS necessary. Will create an ISSUE

Comment (HLWeil) [reply src/ARCtrl/Templates/Templates.Json.fs:111]: #206

File: src/ARCtrl/WebRequest/WebRequest.fs (hunk 2 above)
Comment (HLWeil) [reply src/ARCtrl/WebRequest/WebRequest.fs:10]: Maybe just cut the `My`. Already `NodeJS.stuff` would feel a bit more concise.

File: src/ARCtrl/Templates/Templates.Json.fs [hunk 4]
```diff
@@ -102,4 +103,42 @@ module Template =
...
             )
-        )
\ No newline at end of file
+        )
+
+module Templates =
+
+    let encode (templateList: (string*Template) []) =
+        templateList
+        |> Array.toList
```
Comment (HLWeil) [inline src/ARCtrl/Templates/Templates.Json.fs:112]: Replace by Seq.toList? Would make the function more broadly applicable

File: src/ARCtrl/Templates/Templates.Json.fs [hunk 5]
```diff
@@ -102,4 +103,42 @@ module Template =
...
+            Encode.toString spaces (Template.encode this)
+
+        static member toJson(?spaces: int) =
+            fun (template: Template) ->
+                template.ToJson(?spaces=spaces)
+
+        static member fromJson(jsonString: string) =
+            Decode.fromString Template.decode jsonString
+
+        static member GetTemplates(callback: Map<string,Template> -> unit,?url: string) =
```
Comment (HLWeil) [inline src/ARCtrl/Templates/Templates.Json.fs:136]: Can you even retrieve the templates by this? Or is the outputtype `Unit`

File: src/ARCtrl/WebRequest/WebRequest.fs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+    let downloadFile url callback =
+        fetch url []
+        |> Promise.bind (fun res -> res.text()) // get the result
+        |> Promise.map (fun txt -> // bind the result to make further operation
+            callback txt
+        )
+        |> Async.AwaitPromise
+
+let downloadFile url (callback: string -> unit)=
```
Comment (HLWeil) [inline src/ARCtrl/WebRequest/WebRequest.fs:35]: I get the callback stuff now. But we should at least provide another function for acturally retrieving the object.

**Review by HLWeil — APPROVED** [review]

**Pull request merged**